commit 3acc4522d89e0a326db69e9d0afaad8cf763a54c
Author: Jaegeuk Kim <jaegeuk@kernel.org>
Date:   Sun Oct 25 07:35:47 2020 -0700

    f2fs: call f2fs_get_meta_page_retry for nat page
    
    When running fault injection test, if we don't stop checkpoint, some stale
    NAT entries were flushed which breaks consistency.
    
    Fixes: 86f33603f8c5 ("f2fs: handle errors of f2fs_get_meta_page_nofail")
    Reviewed-by: Chao Yu <yuchao0@huawei.com>
    Signed-off-by: Jaegeuk Kim <jaegeuk@kernel.org>

diff --git a/fs/f2fs/node.c b/fs/f2fs/node.c
index d5d8ce077f29..42394de6c7eb 100644
--- a/fs/f2fs/node.c
+++ b/fs/f2fs/node.c
@@ -109,7 +109,7 @@ static void clear_node_page_dirty(struct page *page)
 
 static struct page *get_current_nat_page(struct f2fs_sb_info *sbi, nid_t nid)
 {
-	return f2fs_get_meta_page(sbi, current_nat_addr(sbi, nid));
+	return f2fs_get_meta_page_retry(sbi, current_nat_addr(sbi, nid));
 }
 
 static struct page *get_next_nat_page(struct f2fs_sb_info *sbi, nid_t nid)